common/misc.go

```diff
@@ -5,35 +5,26 @@ import (
 	"fmt"
 	"net"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 )
 
-func callPowershell(cmds ...string) error {
-	c := []string{"-NonInteractive"}
-	for _, cmd := range cmds {
-		c = append(c, cmd)
-	}
-	return exec.Command("powershell", c...).Run()
-}
-
 func HardResetHNS() error {
 	log.Infoln("Resetting HNS")
 	log.Debugln("Removing NAT")
-	if err := callPowershell("Get-NetNat", "|", "Remove-NetNat"); err != nil {
-		log.Debugln("Could not remove nat network.")
+	if _, _, err := CallPowershell("Get-NetNat", "|", "Remove-NetNat"); err != nil {
+		log.Debugln("Could not remove nat network:", err)
 	}
 	log.Debugln("Removing container networks")
-	if err := callPowershell("Get-ContainerNetwork", "|", "Remove-ContainerNetwork",
+	if _, _, err := CallPowershell("Get-ContainerNetwork", "|", "Remove-ContainerNetwork",
```

Comment:
HardResetHNS is used only in tests, where we want to achieve isolation. I could add some logic here, but there is possibility that it would hinder perfect isolation that we get by just removing everything. I thought about this, but I don't think this is worth the time at this point.